backend/urls.py

```diff
@@ -33,6 +33,10 @@
     path('api/', include(router.urls)),
     path('api/auth/', include('rest_framework.urls', namespace='rest_framework')),
 
+    # Redirect all of these to a login template page to display a message
+    path('accounts/login/', views.get_login_template, name='custom_login_view_accounts'),
+    path('admin/login/', views.get_login_template, name='custom_login_page_admin'),
+    path('admin/logout/', views.get_login_template, name='custom_logout_view_admin'),
```

Comment:
Is there a way to remove these rather than overriding them? Just curious. I saw you could customize the `AdminSite` class, but didn't see how Django itself recommends doing this. https://docs.djangoproject.com/en/4.0/ref/contrib/admin/#adminsite-objects

Reply:
I didn't see any ideas either. this [admin.site.urls](https://github.com/django/django/blob/c627226d05dd52aef59447dcfb29cec2c2b11b8a/django/contrib/admin/sites.py#L317) is a dynamic method that generates urls from get_urls. 

It would be possible to override/extend some of these classesthat but I don't know if that's a better idea than this.

The URLS [are here.](https://github.com/django/django/blob/c627226d05dd52aef59447dcfb29cec2c2b11b8a/django/contrib/admin/sites.py#L269)

There are some others here like /admin/password-change but they redirect to /admin/login first.

I basically [used this suggestion from here](https://stackoverflow.com/a/61946707/3708872) without finding a better one. 

I think the eventual fix would be that we write our custom admin, I just wanted to remove any potential backdoors.

Reply:
There were some other ideas here. Rather than define a template with a nice message we **could** use the view [django.views.defaults.page_not_found](https://docs.djangoproject.com/en/4.0/ref/views/#the-404-page-not-found-view).

I also didn't try if it's possible to do something like add the path
`    path('admin/', admin.site.urls),`

But then remove these paths from the list?

Reply:
I think what you're doing is probably OK. Changing the name of the view was probably a good call.

Reply:
I looked into this and couldn't figure out anything great.

I have an idea of running it through a configurable filter. This **feels** like is should work. This produces a filtered tuple that looks like we can pass it into path. This can for sure be improved. 

```
from django.urls import URLPattern

def exclude_pattern(var, patterns: list):
    # Just ignore these if they don't have a name because it's probably a URLResolver
    if type(var) is URLPattern and var.name in patterns:
        return False
    return True

(list(filter(partial(exclude_pattern, patterns=["login", "logout", "password_change", "password_change_done"]), admin.site.urls[0])), admin.site.urls[1], admin.site.urls[2])
```

This looks like it's always a pattern of  [(list, tuple)](https://github.com/django/django/blob/c627226d05dd52aef59447dcfb29cec2c2b11b8a/django/urls/conf.py#L71).


I didn't like the default look of the page_not_found and it's not **really** a 404.

Reply:
So this seemed like a good idea but removing these results in an error when other code in admin tries to reverse these

`Reverse for 'login' not found. 'login' is not a valid view function or pattern name.`

[This solution to just wrap the method login in a login_required](https://stackoverflow.com/a/13186337/3708872) seems to work fine.